ruby-challenges.rb

```diff
@@ -10,19 +10,60 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def contains_letter (array, string)
+    array.select {|value| value.include? string }
+end
+
+p contains_letter beverages_array, letter_o
+p contains_letter beverages_array, letter_t
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
+# this one felt like it needed some pseudocoding - I used .to_a to turn the hash into an array, then, on the array I used the flatten method to remove the inside arrays, however this left behind the keys (symbols). I used the select method to filter the array to only strings, and then alphabetized it using .sort.
+
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+def alphabetize hash
+new_arr = hash.to_a
```

Comment:
Nice work!  To simplify this even further, take a look at the Ruby built in method .values